rslib/src/import_export/text/import.rs

```diff
@@ -83,6 +89,36 @@ struct DeckIdsByNameOrId {
     default: Option<DeckId>,
 }
 
+/// Notes in the collection indexed by notetype, checksum and optionally deck.
+/// With deck, a note will be included in as many entries as its cards
+/// have different original decks.
+#[derive(Debug)]
+enum ExistingChecksums {
+    WithoutDeck(HashMap<(NotetypeId, u32), Vec<NoteId>>),
+    WithDeck(HashMap<(NotetypeId, u32, DeckId), Vec<NoteId>>),
```

Comment:
I'd originally assumed we'd just ignore any cards outside of the currently-selected deck, but of course that doesn't work for users who have per-row decks specified. This implementation seems like a better approach.